### PR TITLE
Pcs cli simulation merge

### PR DESCRIPTION
**Pull request by hathind-ms**
# Type of change? 

- [ ] Bug fix
- [ ] New feature
- [x ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
Merging custom Simulation branch and adding CLI support for Device Simulation
...

**Checklist:**

- [x ] All tests passed
- [x ] The code follows the code style and conventions of this project
- [x ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/deploymentmanager.ts [hunk 1]
```diff
@@ -227,19 +229,23 @@ export class DeploymentManager implements IDeploymentManager {
...
-                        console.log(chalk.yellow(`Please start streaming jobs mannually once local containers are running: ${jobUrl} `));
+                      const jobUrl =
+                        `${resourceGroupUrl}/providers/Microsoft.StreamAnalytics/streamingjobs/${outputJobName.value}`;
+                      console.log(chalk.yellow(
+                        `Please start streaming jobs mannually once local containers are running: ${jobUrl} `));
                     } else {
-                        deployUI.start(`Waiting for streaming jobs to be started, this could take up to a few minutes.`);
-                        return this.waitForStreamingJobsToStart(answers.solutionName, outputJobName.value);
+                      deployUI.start(`Waiting for streaming jobs to be started, this could take up to a few minutes.`);
+                      return this.waitForStreamingJobsToStart(answers.solutionName, outputJobName.value);
```
Comment (ppathan) [inline src/deploymentmanager.ts]: tab spacing is wrong it should be 4.

File: src/deploymentmanager.ts [hunk 2]
```diff
@@ -227,19 +229,23 @@ export class DeploymentManager implements IDeploymentManager {
...
                     }
+                  }
                 }
                 return Promise.resolve(true);
-            })
+              })
```
Comment (ppathan) [inline src/deploymentmanager.ts]: nit: extra space.

File: src/index.ts [hunk 3]
```diff
@@ -79,17 +79,18 @@ let answers: Answers = {};
...
     .option('-s, --sku <sku>', 'SKU Type (only for Remote Monitoring): basic, standard, or local', /^(basic|standard|local)$/i, 'basic')
     .option('-e, --environment <environment>',
             'Azure environments: AzureCloud or AzureChinaCloud',
             /^(AzureCloud|AzureChinaCloud)$/i, 'AzureCloud')
-    .option('-r, --runtime <runtime>', 'Microservices runtime: dotnet or java', /^(dotnet|java)$/i, 'dotnet')
+    .option('-r, --runtime <runtime>', 'Microservices runtime (only for Remote Monitoring): dotnet or java', /^(dotnet|java)$/i, 'dotnet')
     .option('--servicePrincipalId <servicePrincipalId>', 'Service Principal Id')
     .option('--servicePrincipalSecret <servicePrincipalSecret>', 'Service Principal Secret')
     .option('--versionOverride <versionOverride>', 'Current accepted value is "master"')
+    .option('--dockerTagOverride <dockerTagOverride>', 'Override value for docker image tag')
```
[the remark this answers is not in the document]
Comment (hathind-ms) [reply src/index.ts]: updated.

File: src/deploymentmanager.ts (hunk 1 above)
Comment (hathind-ms) [reply src/deploymentmanager.ts]: fixed

File: src/deploymentmanager.ts (hunk 2 above)
Comment (hathind-ms) [reply src/deploymentmanager.ts]: removed extra space

**Pull request merged**